lib/hijack/http.js

```diff
@@ -1,29 +1,83 @@
+/* eslint-disable prefer-rest-params */
+import { haveAsyncCallback } from '../utils';
+
 if (Package['http']) {
   const HTTP = Package['http'].HTTP;
-
-  var originalCall = HTTP.call;
+  const library = 'meteor/http';
+  const originalCall = HTTP.call;
 
   HTTP.call = function (method, url) {
-    var kadiraInfo = Kadira._getInfo();
-    if (kadiraInfo) {
-      var eventId = Kadira.tracer.event(kadiraInfo.trace, 'http', { method: method, url: url });
+    const tracer = Kadira.tracer;
+    const kadiraInfo = Kadira._getInfo();
+
+    const event = kadiraInfo ? tracer.event(kadiraInfo.trace, 'http', {
+      method,
+      url,
+      library,
+    }) : null;
+
+    if (!event) {
+      return originalCall.apply(this, arguments);
+    }
+
+    try {
+      const response = originalCall.apply(this, arguments);
+
+      // If the user supplied an asyncCallback,
+      // we don't have a response object and it handled asynchronously.
+      // We need to track it down to prevent issues like: #3
+      const endOptions = haveAsyncCallback(arguments) ? { async: true, library } : { statusCode: response.statusCode, library };
+
+      tracer.eventEnd(kadiraInfo.trace, event, endOptions);
+
+      return response;
+    } catch (ex) {
+      tracer.eventEnd(kadiraInfo.trace, event, { err: ex.message });
+
+      throw ex;
+    }
+  };
+}
+
+if (Package['fetch']) {
+  const library = 'meteor/fetch';
+  const originalCall = Package['fetch'].fetch;
+  const Request = Package['fetch'].Request;
+
+  Package['fetch'].fetch = function (url, opts) {
+    const request = new Request(url, opts);
+    console.log(`call fetch:`,request.url);
+    const tracer = Kadira.tracer;
+    const kadiraInfo = Kadira._getInfo();
+
+    const event = kadiraInfo ? tracer.event(kadiraInfo.trace, 'http', {
+      method: request.method,
+      url: request.url,
+      library,
+    }) : null;
+
+    if (!event) {
+      return originalCall.apply(this, arguments);
     }
 
     try {
-      var response = originalCall.apply(this, arguments);
-
-      //if the user supplied an asynCallback, we don't have a response object and it handled asynchronously
-      //we need to track it down to prevent issues like: #3
-      var endOptions = HaveAsyncCallback(arguments) ? { async: true } : { statusCode: response.statusCode };
-      if (eventId) {
-        Kadira.tracer.eventEnd(kadiraInfo.trace, eventId, endOptions);
-      }
+      const response = originalCall.apply(this, arguments);
+
+      const endOptions = { library };
```

Comment:
@zodern Should I add an `async` flag? How it works?

Reply:
Fetch is always async. You can either not have the async field, or have it always be true. Either option should be fine.